File: Front-End/src/services/notification.service.ts
import http from "@/utils/http";

export const notificationApi = {
  getAllNotifications: async (page = 1, limit = 3) => {
    const response = await http.get("/notifications/me", {
      params: { page, limit },
    });
    return response.data;
  },

  countUnreadNotificationsByEmployee: async () => {
    const response = await http.get("/notifications/me/unread/count");
    return response.data.data;
  },

  updateNotificationToRead: async (notificationId: number) => {
    const response = await http.put(`/notifications/me/${notificationId}/read`);
    return response.data;
  },

  updateAllNotificationsToReadByEmployee: async () => {
    const response = await http.put("/notifications/me/read");
    return response.data;
  },
};

File: Back-End-M/src/main/java/com/attendance/fpt/services/NotificationService.java
package com.attendance.fpt.services;

import com.attendance.fpt.entity.Employee;
import com.attendance.fpt.model.response.NotificationResponse;
import com.attendance.fpt.model.response.ResponseWithPagination;

import java.util.List;

public interface NotificationService {
    void sendNotification(Employee employee, String message);

    ResponseWithPagination<List<NotificationResponse>> getAllNotifications(int page, int limit);

    long countUnreadNotificationsByEmployee();

    void updateNotificationToRead(Long notificationId);

    void updateAllNotificationsToReadByEmployee();
}
